ogr/services/github.py

```diff
@@ -66,8 +99,27 @@ def get_branches(self) -> List[str]:
     def get_description(self) -> str:
         return self.github_repo.description
 
-    def get_fork(self) -> Optional[GitProject]:
-        raise NotImplementedError
+    def get_fork(self, create: bool = True) -> Optional["GithubProject"]:
+        """
+        Provide GithubProject instance of a fork of this project.
+
+        Returns None if this is a fork.
+
+        :param create: create a fork if it doesn't exist
+        :return: instance of GithubProject
+        """
+        if self.is_fork:
+            return None
+        f = self.is_forked()
+        if not f:
+            if create:
+                return self.fork_create()
+            else:
+                raise OgrException(
+                    f"Fork of {self.github_repo.full_name}"
+                    " does not exist and we were asked not to create it."
```

Comment:
I would expect `None` in this case.

(In Gitlab context, the forking can be nested and "is_fork == not is_forked" is not always true.)

Reply:
ok

Reply:
I guess you can also do forks of forks on github